cmd/natsdumper/csvwriter/csv_writer.go

```diff
@@ -0,0 +1,197 @@
+package csvwriter
+
+import (
+	"encoding/json"
+	"fmt"
+	"math/rand"
+	"os"
+	"reflect"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/nats-io/go-nats"
+	"github.com/regner/albiondata-client/lib"
+)
+
+var (
+	SaveLocally     bool
+	SaveLocallyPath string
+)
+
+// Used to generate a session ID for unique files
+const letterBytes = "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789"
+const (
+	letterIdxBits = 6                    // 6 bits to represent a letter index
+	letterIdxMask = 1<<letterIdxBits - 1 // All 1-bits, as many as letterIdxBits
+	letterIdxMax  = 63 / letterIdxBits   // # of letter indices fitting in 63 bits
+)
+
+var src = rand.NewSource(time.Now().UnixNano())
+var sessionID string = randStringBytesMaskImprSrc(16)
+
+type marketOrderStruct struct {
+	Id               int    `json:"Id"`
+	ItemTypeId       string `json:"ItemTypeId"`
+	LocationId       int    `json:"LocationId"`
+	QualityLevel     int    `json:"QualityLevel"`
+	EnchantmentLevel int    `json:"EnchantmentLevel"`
+	UnitPriceSilver  int    `json:"UnitPriceSilver"`
+	Amount           int    `json:"Amount"`
+	AuctionType      string `json:"AuctionType"`
+	Expires          string `json:"Expires"`
+}
+
+type goldPricesUploadStruct struct {
+	Prices     []int `json:"Prices"`
+	Timestamps []int `json:"Timestamps"`
+}
+
+func randStringBytesMaskImprSrc(n int) string {
```

Comment:
This looks like a simpler way to generate a UUID: https://play.golang.org/p/4FkNSiUDMg